google/cloud/bigquery/dbapi/_helpers.py

```diff
@@ -113,6 +114,135 @@ def array_to_query_parameter(value, name=None, query_parameter_type=None):
     return bigquery.ArrayQueryParameter(name, array_type, value)
 
 
+complex_query_parameter_parse = re.compile(
+    r"""
+    \s*
+    (ARRAY|STRUCT|RECORD)  # Type
+    \s*
+    <([A-Z0-9<> ,]+)>      # Subtype(s)
+    \s*$
+    """,
+    re.IGNORECASE | re.VERBOSE,
+    ).match
+parse_struct_field = re.compile(
+    r"""
+    (?:(\w+)\s+)    # field name
+    ([A-Z0-9<> ,]+)  # Field type
+    $""", re.VERBOSE | re.IGNORECASE).match
+
+
+def split_struct_fields(fields):
+    fields = fields.split(',')
+    while fields:
+        field = fields.pop(0)
+        while fields and field.count('<') != field.count('>'):
+            field += ',' + fields.pop(0)
```

Comment:
How many fields can we normally expect? Enough for `pop(0)`, which is an `O(N)` operation, to be noticeable?

We might want to process a reversed list of of fields instead, and pop from the right end, or use a `deque`?

Reply:
The number of fields is the number of fields in a struct, which is usually small.

Reply:
I added  `list(reversed(fields)).